src/back/glsl/mod.rs

```diff
@@ -1455,6 +1470,22 @@ impl<'a, W: Write> Writer<'a, W> {
                                 }
                             }
                         }
+
+                        if let back::FunctionType::EntryPoint(ep_index) = ctx.ty {
+                            if self.module.entry_points[ep_index as usize].stage
+                                == crate::ShaderStage::Vertex
+                                && self
+                                    .options
+                                    .writer_flags
+                                    .contains(WriterFlags::ADJUST_COORDINATE_SPACE)
+                            {
+                                writeln!(
+                                    self.out,
+                                    "gl_Position.z = gl_Position.z * 2.0 - gl_Position.w;",
```

Comment:
I don't know much about the coordinate systems but from what I know and read about whilst searching for reviewing this, first shouldn't we flip the Y axis? And second from what I read here https://matthewwellings.com/blog/the-new-vulkan-coordinate-system/ it says that the code for doing this should be
```
gl_Position.z = (gl_Position.z + gl_Position.w) / 2.0;
```
Once again I'll say that I don't really know much about this, so feel free to ignore this comment

Reply:
I don't think the Y needs to be flipped. Otherwise, we'd see this right away in the GL backend.
The transformation you posted is for the *opposite*: going from (-1,1) to (0,1).